### DISCUSSION TITLE
Custom Auth Login Fields

### DISCUSSION THREAD

**Discussion by pslxx** [filamentphp/filament#8036] [Ideas]
I would like to have the ability to configure the validation field for login. Every time I use it, I have to customize the login page and template because in many cases, we do not use email as the username. Instead, we often use usernames or mobile numbers. It would be great to have a place where this can be set.

3Q

**Comment by pslxx** [comment]
Due to the reasons mentioned above, it is unfortunate that the "make:filament-user" command cannot be used.

**Comment by pslxx** [comment]
When I create multiple panels, such as an admin panel, an agent panel, and a CRM panel, they use different data models and have different login fields. I would like to be able to easily customize the login fields for each of them.

**Comment by danharrin** [comment]
The `make:filament-user` command is just to help get you started quickly. You can customize the login and registration pages by overriding them as mentioned in the docs, which allow you to change the fields.

**Comment by pslxx** [comment]
Yes! I have been doing it this way every time since version 2.0. I have already worked on over a dozen projects using this approach, but I've had enough.
^^

**Reply by danharrin** [reply]
Extract your custom pages into a package that can be required across multiple projects then
